File: python/n167_twoNums.py
# https://leetcode-cn.com/problems/two-sum-ii-input-array-is-sorted/

class Solution:
    def twoSum(self, numbers, target: int):
        p_h = 0
        p_t = len(numbers) - 1
        while p_h < p_t:
            if numbers[p_h] + numbers[p_t] < target:
                p_h += 1
            elif numbers[p_h] + numbers[p_t] > target:
                p_t -= 1
            else:
                return (p_h+1, p_t+1)
        return None


if __name__ == "__main__":
    pass

File: cpp/n167_twoNums.cpp
// https://leetcode-cn.com/problems/two-sum-ii-input-array-is-sorted/
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int iter_h = 0;
        int iter_t = numbers.size() - 1;
        while (iter_h < iter_t){
            if (numbers[iter_h] + numbers[iter_t] == target) return {iter_h+1, iter_t+1};
            if (numbers[iter_h] + numbers[iter_t] < target) iter_h += 1;
            if (numbers[iter_h] + numbers[iter_t] > target) iter_t -= 1;
        }
        return {};
    }
};

int main(){
    return 0;
}